Cabal/doc/nix-local-build.markdown

```diff
@@ -0,0 +1,364 @@
+
+   The global package store is `~/.cabal/store`; if you need to clear
+   your store for whatever reason (e.g., to reclaim disk space or
+   because the global store is corrupted), deleting this directory is
+   safe (`new-build` will just rebuild everything it needs on its next
+   invocation).
+
+This split motivates some of the UI choices for Nix-style local
+build commands.  For example, flags passed to `cabal new-build` are
+only applied to *local* packages, so that adding a flag to `cabal
+new-build` doesn't necessitate a rebuild of *every* transitive
+dependency in the global package store.
+
+# Commands #
+
+We now give an in-depth description of all the commands, describing
+the arguments and flags they accept.
+
+## cabal new-configure ##
+
+`cabal new-configure` takes a set of arguments and
+writes a `cabal.project.local` file based on the flags passed to this
+command.  `cabal new-configure FLAGS; cabal new-build` is roughly equivalent
+to `cabal new-build FLAGS`, except that with `new-configure` the flags
+are persisted to all subsequent calls to `new-build`.
+
+`cabal new-configure` is intended to be a convenient way to write out a
+`cabal.project.local` for simple configurations; e.g., `cabal
+new-configure -w ghc-7.8` would ensure that all subsequent builds with
+`cabal new-build` are performed with the compiler `ghc-7.8`.  For
+more complex configuration, we recommend writing the
+`cabal.project.local` file directly (or placing it in `cabal.project`!)
+
+`cabal new-configure` inherits options from `Cabal`.
+semantics:
+
+* Any flag accepted by `./Setup configure`.
+
+* Any flag accepted by `cabal configure` beyond `./Setup configure`,
+  namely `--cabal-lib-version`, `--constraint`, `--preference` and
+  `--solver.`
+
+* Any flag accepted by `cabal install` beyond `./Setup configure`.
+
+* Any flag accepted by `./Setup haddock`.
+
+The options of all of these flags apply only to *local* packages
+in a project; this behavior is different than that of `cabal install`,
+which applies flags to every package that would be built.  The
+motivation for this is to avoid an innocuous addition to the flags
+of a package resulting a rebuild of every package in the store
+(which might need to happen if a flag actually applied to every
+transitive dependency).  To apply options to an external package,
+use a `package` stanza in a `cabal.project` file.
+
+## cabal new-build ##
+
+`cabal new-build` takes a set of targets and builds them.  It
+automatically handles building and installing any dependencies
+of these targets.
+
+A target can take any of the following forms:
+
+* A package target: `package`, which specifies that all
+  enabled components of a package to be built.  By default,
+  test suites and benchmarks are *not* enabled, unless
+  they are explicitly requested (e.g., via `--enable-tests`.)
+
+* A component target: `[package:][ctype:]component`, which
+  specifies a specific component to be built.
```

Comment:
Perhaps worth explicitly pointing out that this can be a test suite or benchmark, as a counterpoint to the fact that tests are not included when you ask to build a package as a whole.